crates/indexer/src/models/token_models/v2_token_activities.rs

```diff
@@ -0,0 +1,226 @@
+    pub type_: String,
+    pub from_address: Option<String>,
+    pub to_address: Option<String>,
+    pub token_amount: BigDecimal,
+    pub before_value: Option<String>,
+    pub after_value: Option<String>,
+    pub entry_function_id_str: Option<String>,
+    pub token_standard: String,
+    pub is_fungible_v2: Option<bool>,
+    pub transaction_timestamp: chrono::NaiveDateTime,
+}
+
+/// A simplified TokenActivity (excluded common fields) to reduce code duplication
+struct TokenActivityHelperV1 {
+    pub token_data_id_struct: TokenDataIdType,
+    pub property_version: BigDecimal,
+    pub from_address: Option<String>,
+    pub to_address: Option<String>,
+    pub token_amount: BigDecimal,
+}
+
+/// A simplified TokenActivity (excluded common fields) to reduce code duplication
+struct TokenActivityHelperV2 {
+    pub from_address: Option<String>,
+    pub to_address: Option<String>,
+    pub token_amount: BigDecimal,
+    pub before_value: Option<String>,
+    pub after_value: Option<String>,
+}
+
+impl TokenActivityV2 {
+    pub fn get_v2_nft_from_parsed_event(
+        event: &APIEvent,
+        txn_version: i64,
+        txn_timestamp: chrono::NaiveDateTime,
+        event_index: i64,
+        entry_function_id_str: &Option<String>,
+        token_v2_metadata: &TokenV2AggregatedDataMapping,
+    ) -> anyhow::Result<Option<Self>> {
+        let event_type = event.typ.to_string();
+        if let Some(token_event) =
+            &V2TokenEvent::from_event(event_type.as_str(), &event.data, txn_version)?
+        {
+            let event_account_address =
+                standardize_address(&event.guid.account_address.to_string());
+            // burn and mint events are attached to the collection. The rest should be attached to the token
+            let token_data_id = match token_event {
+                V2TokenEvent::MintEvent(inner) => inner.get_token_address(),
+                V2TokenEvent::BurnEvent(inner) => inner.get_token_address(),
+                V2TokenEvent::TransferEvent(inner) => inner.get_object_address(),
+                _ => event_account_address.clone(),
+            };
+
+            if let Some(metadata) = token_v2_metadata.get(&token_data_id) {
+                let object_core = &metadata.object;
+                let token_activity_helper = match token_event {
+                    V2TokenEvent::MintEvent(_) => TokenActivityHelperV2 {
+                        from_address: Some(object_core.get_owner_address()),
+                        to_address: None,
+                        token_amount: BigDecimal::one(),
+                        before_value: None,
+                        after_value: None,
+                    },
+                    V2TokenEvent::TokenMutationEvent(inner) => TokenActivityHelperV2 {
+                        from_address: Some(object_core.get_owner_address()),
+                        to_address: None,
+                        token_amount: BigDecimal::zero(),
```

Comment:
nit: should this be 1? because 1 token is being mutated. I guess a larger question is what is token_amount supposed to represent?

Reply:
the option here are really 0 or null b/c this is modifying token data level properties (e.g. changing URI) so in the case of fungible tokens we can't really get the right amount if that makes sense.